olive/platform_sdk/qualcomm/runner.py

```diff
@@ -0,0 +1,94 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+import time
+
+from olive.common.utils import run_subprocess
+from olive.platform_sdk.qualcomm.snpe.env import SNPEAndroidEnv, SNPESDKEnv
+from olive.platform_sdk.qualcomm.utils.raw_adb import run_adb_command
+
+logger = logging.getLogger(__name__)
+
+
+class SDKRunner:
+    def __init__(
+        self,
+        platform,
+        cmd: str,
+        dev: bool = False,
+        runs: int = 1,
+        sleep: int = 0,
+        log_error: bool = True,
+        android_target: str = None,
+    ):
+        self.platform = platform
+        self.cmd = cmd
+        self.dev = dev
+        self.runs = runs
+        self.sleep = sleep
+        self.log_error = log_error
+        self.android_target = android_target
+
+    def runner_env(self):
```

Comment:
Should we cache the result of this method so that it doesn't have to go through the whole environment creation every time? Maybe using something like `functools.lru_cache`? https://docs.python.org/3/library/functools.html#functools.lru_cache

Reply:
Actually, nevermind. Just saw this is used to only run the command once.

Reply:
Aha, yes, it is call once in runtime. 
Another thing, we should avoid use lru_cache to decorate instance method.
https://rednafi.com/python/lru_cache_on_methods/